### ISSUE TITLE
Docker container for 0.9.0 version failed in early July (July 3 -5)

### ISSUE THREAD

**Issue by gregelin** [GovReady/govready-q#659]
Docker container for 0.9.0 failed in early July. Browser displayed "Internal Server Error".

## Diagnosed issue

Django module (app) `testmocking` did not contain an `__init__.py` file.

Django requires empty `__init__.py` file in each app.

Reproduced issue by building and running local docker container.

## Resolution

Copy `__init__.py` from another existing application into `testmocking`

## Debugging
Look at logs:
```
# commands
docker exec -it <container_id> /bin/bash
ls /var/logs
cat /var/log/application-stderr---supervisor-<XXXX>.log

# log result
[uWSGI] getting INI configuration from /tmp/uwsgi.ini
*** Starting uWSGI 2.0.18 (64bit) on [Sat Jul  6 13:45:00 2019] ***
compiled with version: 4.8.5 20150623 (Red Hat 4.8.5-36) on 06 July 2019 13:24:40
os: Linux-4.9.125-linuxkit #1 SMP Fri Sep 7 08:20:28 UTC 2018
nodename: a3bef0b3d393
machine: x86_64
clock source: unix
pcre jit disabled
detected number of CPU cores: 2
current working directory: /usr/src/app
detected binary path: /usr/bin/uwsgi
your memory page size is 4096 bytes
detected max file descriptor number: 1048576
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uWSGI http bound on 0.0.0.0:8000 fd 4
uwsgi socket 0 bound to TCP address 127.0.0.1:38653 (port auto-assigned) fd 3
Python version: 3.6.8 (default, May  2 2019, 20:40:44)  [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
*** Python threads support is disabled. You can enable it with --enable-threads ***
Python main interpreter initialized at 0x1313120
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 364600 bytes (356 KB) for 4 cores
*** Operational MODE: preforking ***
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/django/apps/config.py", line 107, in create
    entry = module.default_app_config
AttributeError: module 'testmocking' has no attribute 'default_app_config'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "siteapp/wsgi.py", line 5, in <module>
    django_application = get_wsgi_application()
  File "/usr/lib64/python3.6/site-packages/django/core/wsgi.py", line 12, in get_wsgi_application
    django.setup(set_prefix=False)
  File "/usr/lib64/python3.6/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/lib64/python3.6/site-packages/django/apps/registry.py", line 89, in populate
    app_config = AppConfig.create(entry)
  File "/usr/lib64/python3.6/site-packages/django/apps/config.py", line 110, in create
    return cls(entry, module)
  File "/usr/lib64/python3.6/site-packages/django/apps/config.py", line 40, in __init__
    self.path = self._path_from_module(app_module)
  File "/usr/lib64/python3.6/site-packages/django/apps/config.py", line 73, in _path_from_module
    "with a 'path' class attribute." % (module, paths))
django.core.exceptions.ImproperlyConfigured: The app module <module 'testmocking' (namespace)> has multiple filesystem locations (['./testmocking', '/usr/src/app/testmocking']); you must configure this app with an AppConfig subclass with a 'path' class attribute.
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 40)
spawned uWSGI worker 1 (pid: 44, cores: 1)
spawned uWSGI worker 2 (pid: 45, cores: 1)
spawned uWSGI worker 3 (pid: 46, cores: 1)
spawned uWSGI worker 4 (pid: 47, cores: 1)
*** Stats server enabled on 127.0.0.1:9191 fd: 18 ***
spawned uWSGI http 1 (pid: 48)
--- no python application found, check your startup logs for errors ---
[pid: 44|app: -1|req: -1/1] 172.17.0.1 () {38 vars in 734 bytes} [Sat Jul  6 13:45:03 2019] GET / => generated 21 bytes in 0 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 0)
```

**Comment by gregelin** [comment]
@ethankaminski FYI, see above. All Django modules/apps need a `__init__.py` file at their root.

This only caused a problem when running the containerized version.

**Comment by ethankaminski** [comment]
@gregelin Got it, thanks for the heads-up.

**Comment by gregelin** [comment]
@ethankaminski Can we close this issue of including an empty `__init__.py` in testmocking?

**Comment by ethankaminski** [comment]
@gregelin fixed on `0.9.0.dev`, not fixed on `master` yet (but PR #672 includes the fix for 0.8.6). So, it's fixed on 0.9.0, correct; and there's an open PR which fixes it on 0.8.6.

**Comment by gregelin** [comment]
fixed.
